crates/rust-analyzer/src/to_proto.rs

```diff
@@ -864,6 +864,22 @@ pub(crate) fn snippet_text_document_ops(
                 rename_file,
             )))
         }
+        FileSystemEdit::MoveDir { src, src_id, dst } => {
+            let old_uri = snap.anchored_path(&src);
+            let new_uri = snap.anchored_path(&dst);
+            let mut rename_file =
+                lsp_types::RenameFile { old_uri, new_uri, options: None, annotation_id: None };
+            if let Some(src_id) = src_id {
+                if snap.analysis.is_library_file(src_id).ok() == Some(true)
+                    && snap.config.change_annotation_support()
+                {
+                    rename_file.annotation_id = Some(outside_workspace_annotation_id())
```

Comment:
the `src_id` field is added for this annotation_id, not sure if this is the right way